xcdat/dataset.py

```diff
@@ -434,14 +434,37 @@ def _parse_xml_for_nc_glob(xml_path: Union[str, pathlib.Path]) -> str:
     root = tree.getroot()
 
     dir_attr = root.attrib.get("directory")
-    if dir_attr is None:
+    if dir_attr == "":
+        dir_attr = "."
+    elif dir_attr is None:
         raise KeyError(
             f"The XML file ({xml_path}) does not have a 'directory' attribute "
             "that points to a directory of `.nc` dataset files."
         )
 
-    glob_path = dir_attr + "/*.nc"
+    filemap_attr = root.attrib.get("cdms_filemap")
+    if filemap_attr is not None:
+        file_name_list = sorted(
+            list(
+                set(
+                    [
+                        x
+                        for x in filemap_attr.replace("[", " ")
+                        .replace("]", " ")
+                        .replace(",", " ")
+                        .split(" ")
+                        if ".nc" in x
+                    ]
+                )
+            )
+        )
+        glob_path = list()
+        for file_name in file_name_list:
+            glob_path.append(os.path.join(dir_attr, file_name))
```

Comment:
Tried to make this more succint, readable, and added comments. 

```suggestion
        # remove brackets and commas from filemap_attr
        fma = filemap_attr.replace("[", " ").replace("]", " ").replace(",", " ")
        # split string filemap_attr up (by spaces)
        fm_values = fma.split(" ")
        # keep values that are nc files (and remove white space)
        file_name_list = [fn.replace(" ", "") for fn in fm_values if ".nc" in fn]
        # create empty list
        glob_path = list()
        # combine directory and file names
        for file_name in file_name_list:
            glob_path.append(os.path.join(dir_attr, file_name))
```

Reply:
@pochedls this is great! Thank you for the suggestion!